lib/gyp/generator/ninja/index.js

```diff
@@ -410,6 +416,10 @@ Ninja.prototype.actions = function actions() {
 
     res = res.concat(outputs);
 
+    if (action.process_outputs_as_sources === '1') {
+      this.targetDict.sources = (this.targetDict.sources || []).concat(action.outputs);
```

Comment:
Like ensuring that the `sources` is an Array and then doing `concat` on a second line.